cmake/options.cmake

```diff
@@ -184,3 +184,23 @@ set(DNNL_USE_CLANG_SANITIZER "" CACHE STRING
     This feature is experimental and is only available on Linux.")
 
 option(_DNNL_USE_MKL "use BLAS functions from Intel MKL" OFF)
+
+# =====================
+# AArch64 build options
+# =====================
+
+option(DNNL_AARCH64_USE_ARMPL
+    "enable CBLAS interface and link against Arm Performance Libraries
+    if available."
+    OFF)
+
+option(DNNL_AARCH64_USE_OPENBLAS
+    "enable CBLAS interface and link against OpenBLAS
+    if available."
+    OFF)
+
+option(_DNNL_USE_CBLAS "use BLAS functions external BLAS library" OFF)
```

Comment:
I am fine with merging this as-is but it seems like the build scripts would benefit from some unification of the ways how BLAS libraries are handled. It seems like `FindBLAS.cmake` can also detect MKL which duplicates already existing functionality. Maybe it can be removed? Tagging @emfomenk .

Reply:
I tend to agree with @rsdubtso. With the following dirty patch [1] the library seems working fine for me. I also agree with @rsdubtso in this comment: https://github.com/oneapi-src/oneDNN/pull/741#discussion_r435694681.
How about a single option, like `DNNL_BLAS_VENDOR` which could be `ARMPL`, `MKL`, or `OPENBLAS`?
- The option `_DNNL_USE_MKL` will become an alias to `DNNL_BLAS_VENDOR=MKL`.
   - If you consider BLAS support for ARM experimental as well, we can go with `_DNNL_BLAS_VENDOR` for now
- `cmake/MKL.cmake` can be dropped altogether

[1] 
```patch
diff --git a/cmake/blas.cmake b/cmake/blas.cmake
index aa3848a02..735797c09 100644
--- a/cmake/blas.cmake
+++ b/cmake/blas.cmake
@@ -21,19 +21,8 @@ endif()
 set(blas_cmake_included true)
 include("cmake/options.cmake")

-if(DNNL_TARGET_ARCH STREQUAL "AARCH64")
-    if(DNNL_AARCH64_USE_ARMPL)
-        set(_DNNL_USE_CBLAS ON)
-        if(DNNL_CPU_RUNTIME STREQUAL "OMP")
-            set(BLA_VENDOR "Arm_mp")
-        else()
-            set(BLA_VENDOR "Arm")
-        endif()
-    elseif(DNNL_AARCH64_USE_OPENBLAS)
-        set(_DNNL_USE_CBLAS ON)
-        set(BLA_VENDOR "OpenBLAS")
-    endif()
-endif()
+set(BLA_VENDOR "Intel10_64_dyn")
+set(_DNNL_USE_CBLAS ON)

 if(NOT _DNNL_USE_CBLAS)
     return()
@@ -50,8 +39,9 @@ if(BLAS_FOUND)
      set(CMAKE_REQUIRED_FLAGS "${BLAS_COMPILER_FLAGS}")

      # Find and include  accompanying cblas.h
-     get_filename_component(BLAS_LIB_DIR ${BLAS_LIBRARIES} PATH)
-     find_path(BLAS_INCLUDE_DIR cblas.h $ENV{CPATH} ${BLAS_LIB_DIR}/../include)
+     list(GET BLAS_LIBRARIES 0 FIRST_BLAS_LIB)
+     get_filename_component(BLAS_LIB_DIR ${FIRST_BLAS_LIB} PATH)
+     find_path(BLAS_INCLUDE_DIR mkl_cblas.h $ENV{CPATH} ${BLAS_LIB_DIR}/../include ${BLAS_LIB_DIR}/../../include)
      include_directories(${BLAS_INCLUDE_DIR})

      unset(CBLAS_WORKS CACHE)
@@ -64,4 +54,5 @@ if(BLAS_FOUND)
      message(STATUS "Found CBLAS: ${BLAS_LIBRARIES}")
      message(STATUS "CBLAS include path: ${BLAS_INCLUDE_DIR}")
      add_definitions(-DUSE_CBLAS)
+     add_definitions(-DUSE_MKL)
 endif()
```

Reply:
`DNNL_BLAS_VENDOR` looks reasonable to me.